src/resolve/resolved-dns-dnssec.c

```diff
@@ -1233,7 +1240,7 @@ int dnssec_verify_rrset_search(
                                         *ret_rrsig = rrsig;
 
                                 *result = one_result;
-                                return 0;
+                                return nvalidations;
```

Comment:
here you silently cast `size_t` into `int`. 

I think casting here is actually fine, given that DNSSEC_INVALID_MAX is below INT_MAX, but I'd prefer this would be made explicit even if unnecessary, 

i.e. `return (int) nvalidations;`